### PR TITLE
V2.4 backports

### PR DESCRIPTION
**Pull request by jonludlam**
Backport of some changes to the 2.4 branch, most importantly OCaml 5.2 support.

### CODE REVIEW

**Comment by jonludlam** [comment]
I've reverted a change that enables sherlodoc, as it's breaking the benchmark build. Was this put in for the website build?

**Comment by panglesd** [comment]
Yes, it was put in for the website build (@EmileTrotignon).

What was the problem? I can't find it in the benchmark CI user interface: it only shows the log from the last (passing) build...
It might be tricky to have the benchmark working with sherlodoc: sherlodoc is a revdeps of odoc, so I think for this PR it is fine to revert it. For the main branch we might want to solve the problem...

**Comment by panglesd** [comment]
Yes, sherlodoc does not appear in the dependencies of the `odoc-bench` package.

But sherlodoc has odoc as a dependency. So we need to pin odoc to the `dev` version in the `odoc-bench` package, if I understand correctly?!

**Review by panglesd — APPROVED** [review]
Looks good to me!

File: src/xref2/tools.ml [hunk 1]
```diff
@@ -896,7 +896,13 @@ and lookup_datatype_gpath :
...
         (* CoreTypes aren't put into the environment, so they can't be handled by the
               next clause. We just look them up here in the list of core types *)
-        Ok (`FType (name, List.assoc (TypeName.to_string name) core_types))
+        Ok
+          (`FType
+            ( name,
+              Component.Of_Lang.(
+                type_decl (empty ())
+                  (Odoc_model.Predefined.type_of_core_type
+                     (TypeName.to_string name))) ))
```
[the remark this answers is not in the document]
Comment (jonludlam) [reply src/xref2/tools.ml:905]: Thanks! not sure how that happened. I fixed it anyway, just to be tidy.

**Pull request merged**